include/eve/detail/function/simd/common/swizzle_helpers.hpp

```diff
@@ -0,0 +1,78 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2018-2021 Joel FALCOU
+  Copyright 2018-2021 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/meta.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/detail/function/bit_cast.hpp>
+#include <eve/pattern.hpp>
+
+namespace eve::detail
+{
+  //----------------------------------------------------------------------------------------------
+  // Handle Zeroing
+  template<typename Pack, typename Pattern>
+  EVE_FORCEINLINE auto process_zeros( Pack const& v, Pattern p ) noexcept
+  {
+    constexpr auto sz = cardinal_v<Pack>;
+    using type = typename Pack::value_type;
+
+    if constexpr( p.has_zeros(sz) )
+    {
+      // Turn pattern into 0/~0 then mask
+      auto const impl = [=](auto... i)
+                        {
+                          using i_t = as_integer_t<type,unsigned>;
+                          constexpr i_t nz = ~i_t(0), zz = i_t(0);
+                          return Pack(bit_cast((p(i,sz)==-1 ? zz : nz),as_<type>())...);
```

Comment:
I rly rly think that -1 should be a enum somewhere.

Reply:
it is and we need to use it everywhere

Reply:
Still -1